src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -162,20 +162,24 @@ private void getMealById(HttpServletRequest request, HttpServletResponse respons
     private void returnIdOfSimilar(HttpServletRequest request, HttpServletResponse response) throws IOException {
         // Fuction redirect to random page
         // TODO(grenlayk): implement suggestions algorithm here for Product Alpha
+        Long pageId = Long.parseLong(getParameter(request, "id", "0"), 10);
```

Comment:
Please skip the second argument (the `10`) - by default `parseLong` parses the string as a decimal number.